### PR TITLE
Add realm creator via keycloak-config-cli

### PR DESCRIPTION
**Pull request by ingokuba**

### CODE REVIEW

File: templates/realmcreator.yaml [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+    "helm.sh/hook-weight": {{ .hookWeight | default 5 | quote }}
+spec:
+  template:
+    metadata:
+      name: {{ include "keycloak.fullname" . }}-realm-creator
+    spec:
+      restartPolicy: Never
+      containers:
+      - name: realm-creator
+        image: adorsys/keycloak-config-cli
```
Comment (ingokuba) [inline templates/realmcreator.yaml:25]: where is the tag?
and does it make sense to be configurable?

File: README.md [hunk 2]
```diff
@@ -117,3 +117,16 @@ This chart offers the possibility to define jobs that run after the installation
...
+| `realmCreator.enabled`           | Enable the realm creator                                                                                                                 | `false` |
+| `realmCreator.username`          | Specify the username to be used for the Keycloak API                                                                                     | `""`    |
+| `realmCreator.password`          | Password for the Keycloak user                                                                                                           | `""`    |
+| `realmCreator.existingSecret`    | Use an existing secret for Keycloak admin password. This takes precedence over password.                                                 | `""`    |
+| `realmCreator.existingSecretKey` |                                                                                                                                          | `""`    |
+| `realmCreator.options`           | Specify how the realms should be created. See [keycloak-config-cli](https://github.com/adorsys/keycloak-config-cli) for more information | `""`    |
```
Comment (ingokuba) [inline README.md:132]: does this contain the JSON as YAML for the realm to create?

Comment (Feleuxens) [reply README.md]: It does. I clarified it a bit in the README

File: templates/realmcreator.yaml (hunk 1 above)
Comment (Feleuxens) [reply templates/realmcreator.yaml]: It is now configurable with some defaults set

File: README.md [hunk 3]
```diff
@@ -117,3 +117,16 @@ This chart offers the possibility to define jobs that run after the installation
...
+| -------------------------------- | ---------------------------------------------------------------------------------------------------------------------------------------- | ------- |
+| `realmCreator.enabled`           | Enable the realm creator                                                                                                                 | `false` |
+| `realmCreator.username`          | Specify the username to be used for the Keycloak API                                                                                     | `""`    |
+| `realmCreator.password`          | Password for the Keycloak user                                                                                                           | `""`    |
+| `realmCreator.existingSecret`    | Use an existing secret for Keycloak admin password. This takes precedence over password.                                                 | `""`    |
+| `realmCreator.existingSecretKey` |                                                                                                                                          | `""`    |
```
[the remark this answers is not in the document]
Comment (ingokuba) [reply README.md]: good solution

File: values.yaml [hunk 4]
```diff
@@ -305,3 +305,35 @@ ingress:
...
   #    hosts:
   #      - chart-example.local
+
+
+## @section Realm creator parameters
+## @descriptionStart
+## This Helm chart provides the possibility to create a realm after Keycloak was deployed. This is done using [keycloak-config-cli](https://github.com/adorsys/keycloak-config-cli). All options from realmCreator.options are passed to this cli.
+## @descriptionEnd
+
+realmCreator:
```
Comment (ingokuba) [inline values.yaml:315]: more realms! can we have multiple? :innocent:

File: values.yaml [hunk 5]
```diff
@@ -305,3 +305,35 @@ ingress:
...
+  ## @param realmCreator.username Specify the username to be used for the Keycloak API
+  username: admin
+  ## @param realmCreator.password Password for the Keycloak user
+  password: ""
+  ## @param realmCreator.existingSecret Use an existing secret for Keycloak admin password. This takes precedence over password.
+  existingSecret: ""
+  ## @param realmCreator.existingSecretKey
+  existingSecretKey: ""
+  ## @param realmCreator.options Specify how the realms should be created. It contains the JSON as YAML in a string. See [keycloak-config-cli](https://github.com/adorsys/keycloak-config-cli) for more information
+  options: ""
```
Comment (ingokuba) [inline values.yaml:336]: just replace `options` with `realms` array and then check if empty?

File: values.yaml (hunk 4 above)
Comment (Feleuxens) [reply values.yaml:315]: Yes, we can https://github.com/intensiongmbh/keycloak-helm-chart/pull/3/commits/99340edcdd90efd4303b3170db6676b0d1d38b8b

File: values.yaml (hunk 5 above)
Comment (Feleuxens) [reply values.yaml]: ✅  https://github.com/intensiongmbh/keycloak-helm-chart/pull/3/commits/99340edcdd90efd4303b3170db6676b0d1d38b8b

**Comment by ingokuba** [comment]
Nice

**Pull request merged**